src/Symfony/Component/String/Uuid.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class Uuid implements \JsonSerializable
+{
+    public const TYPE_1 = UUID_TYPE_TIME;
+    public const TYPE_3 = UUID_TYPE_MD5;
+    public const TYPE_4 = UUID_TYPE_RANDOM;
+    public const TYPE_5 = UUID_TYPE_SHA1;
+
+    public const VARIANT_NCS = UUID_VARIANT_NCS;
+    public const VARIANT_DCE = UUID_VARIANT_DCE;
+    public const VARIANT_MICROSOFT = UUID_VARIANT_MICROSOFT;
+    public const VARIANT_OTHER = UUID_VARIANT_OTHER;
+
+    private $uuid;
+
+    public function __construct(string $uuid)
+    {
+        self::checkIfExtensionIsLoaded();
+
+        if (!uuid_is_valid($uuid)) {
+            throw new \InvalidArgumentException('The UUID is not valid.');
+        }
+
+        $this->uuid = $uuid;
+    }
+
+    public static function v1(): self
+    {
+        self::checkIfExtensionIsLoaded();
+
+        return new self(uuid_create(UUID_TYPE_TIME));
+    }
+
+    public static function v3(self $uuidNamespace, string $name): self
+    {
+        self::checkIfExtensionIsLoaded();
+
+        return new self(uuid_generate_md5($uuidNamespace->uuid, $name));
+    }
+
+    public static function v4(): self
+    {
+        self::checkIfExtensionIsLoaded();
+
+        return new self(uuid_create(UUID_TYPE_RANDOM));
+    }
+
+    public static function v5(self $uuidNamespace, string $name): self
+    {
+        self::checkIfExtensionIsLoaded();
+
+        return new self(uuid_generate_sha1($uuidNamespace->uuid, $name));
+    }
+
+    public static function fromBinary(string $uuidAsBinary): self
+    {
+        self::checkIfExtensionIsLoaded();
+
+        return new self(uuid_unparse($uuidAsBinary));
+    }
+
+    public function toBinary(): string
+    {
+        return uuid_parse($this->uuid);
+    }
+
+    public function isNull(): bool
+    {
+        return uuid_is_null($this->uuid);
+    }
+
+    public function isEqualsTo(self $other): bool
```

Comment:
`is equals` sounds strange ... `equals()` or `isEqualTo()` ?

Reply:
`equals()` :+1